fivox/eventSource.cpp

```diff
@@ -283,41 +288,27 @@ void EventSource::buildRTree()
 #endif
 }
 
-bool EventSource::load( const uint32_t frame )
+bool EventSource::setFrame( const uint32_t frame )
 {
     if( !isInFrameRange( frame ))
         return false;
 
     const float time  = _getTimeRange().x() + getDt() * frame;
-    return load( time );
+    setTime( time );
+    return true;
 }
 
-bool EventSource::load( const float time )
+void EventSource::setTime( const float time )
 {
-    if( time == _impl->currentTime )
-        return true;
-
-    const ssize_t updatedEvents = _load( time );
-    if( updatedEvents < 0 )
-    {
-        LBERROR << "Timestamp " << time << "ms not loaded, no data or events"
-                << std::endl;
-        return false;
-    }
-
-    LBINFO << "Timestamp " << time << "ms loaded, updated " << updatedEvents
-           << " event(s)" << std::endl;
-
     _impl->currentTime = time;
```

Comment:
Hmmm, this is not checked at all. Why did you split load into separate load and setTime operations?